### PR TITLE
Project 6 - Weather App - Darcy W and Andrea L

### PR DESCRIPTION
**Pull request by darcywinterhill**
Link to my deployed project:
https://weather-site-project.netlify.app/

My reflections on how this week's project turned out:
We built a weather app that we think turned out clean and easy to read.

Things I'd like to have clarified or explained in more detail:
We had some issues with the code working differently on iOS vs Android but we got help and managed to find solutions that worked with our code.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
+Since the functions and the information we fetched work/look differrently on iOS vs Android we needed to put some extra time into finding fixes and implementing them in our code.
+
+If we had more time we probably would have chosen other weather icons, but now we learned how to fetch it from APIs, which is the reason we decided to go with these.
+We would also have looked into how to build a site where you can search any city for weather.
+
 
 ## View it live
 
-Every project should be deployed somewhere. Be sure to include the link to the deployed project so that the viewer can click around and see what it's all about.
+https://weather-site-project.netlify.app/
```
Comment (Aepera) [inline README.md:19]: The weather app is working as supposed on desktop, iPhone and on android phone. Nice feature with the responsive layout and the text appearing only on larger devices.

File: index.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="stylesheet" href="./style.css">
+    <link rel="preconnect" href="https://fonts.gstatic.com">
+    <link href="https://fonts.googleapis.com/css2?family=Comfortaa:wght@700&display=swap" rel="stylesheet"> 
+    <title>Weather App</title>
+</head>
+<body>
+    <section class="container">
+    <div id="containerToday" class="container_today"></div>
+    <div id="containerForecast" class="container_forecast"></div>
```
Comment (Aepera) [inline index.html:15]: These two rows should be intented 😊

File: script.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
+// API's
+const todaysUrl = 'https://api.openweathermap.org/data/2.5/weather?q=Göteborg&units=metric&APPID=947f288ad7c7a6c1279353f3ee6f09d1'
+const forecastUrl = 'https://api.openweathermap.org/data/2.5/forecast?q=Göteborg&units=metric&APPID=947f288ad7c7a6c1279353f3ee6f09d1'
```
Comment (Aepera) [inline script.js:3]: The API keys could also be stored as variables separate from the URL.

File: script.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        // Variables to display weather data
+        const city = json.name.toUpperCase()
+        const cityName = json.name
+        const temp = json.main.temp.toFixed(1)
+        const timezoneOffset = new Date().getTimezoneOffset() * 60
+        const sunrise = new Date((json.sys.sunrise + json.timezone + timezoneOffset) * 1000).toLocaleString(navigator.language, {hour: '2-digit', minute:'2-digit'})
+        const sunset = new Date((json.sys.sunset + json.timezone + timezoneOffset) * 1000).toLocaleString(navigator.language, {hour: '2-digit', minute:'2-digit'})
+        const weatherType = json.weather[0].main
+        const weatherDescription = json.weather[0].description
+        const wIcon = json.weather[0].icon
```
Comment (Aepera) [inline script.js:38]: We like that the variables are declared here outside of the innerHTML, makes the code easier to read and understand.

File: script.js [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+        // HTML weather text for tablet and desktop, also changes bg color depending on weather type
+        switch (weatherType) {
```
Comment (Aepera) [inline script.js:41]: Nice with a switch statement!

File: script.js [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+        <h2>${city}</h2>
+        <p>${weatherDescription}</p>
+        </div>
+        <img src="https://openweathermap.org/img/wn/${wIcon}@2x.png" />
+        </div>
+        <div class="sunrise_sunset">
+        <p>Sunrise: ${sunrise}</p>
+        <p>Sunset: ${sunset}</p>
+        </div>
+        </section>`
```
Comment (Aepera) [inline script.js:89]: Don´t forget to use indentation in innerHTML!

File: script.js [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+    })
+    
+    .catch(error => {
+        containerToday.innerHTML = `
+        <p>${error}</p>`
+    })
+
+// API fetch 5 day forecast
```
Comment (Aepera) [inline script.js:97]: Short and descriptive comments, make it easy to follow the code.

File: script.js [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+        // HTML forecast weather
+        containerForecast.innerHTML += `
+        <section class="forecast">
+        <p>${weekday}</p>
+        <div class="forecast_weather_temp">
+        <div class="forecast_weather_temp">
+        <img class="forecast_icon" src="https://openweathermap.org/img/wn/${wIcon}@2x.png" />
+        <p class="forecast_temp">${forecastTemp}°C</p>
+        </div>
+        </section>`
```
Comment (Aepera) [inline script.js:126]: Indentation! 😊😊

File: style.css [hunk 9]
```diff
@@ -0,0 +1,302 @@
+body {
+    box-sizing: border-box;
+    position: relative;
+    overflow: hidden;
+    height: 100vh;
+    font-family: 'Comfortaa', cursive;
+    background-image: white;
+}
+
+.cloudy {
```
Comment (Aepera) [inline style.css:10]: Nice with a changing background

File: style.css [hunk 10]
```diff
@@ -0,0 +1,302 @@
...
+    background-image: linear-gradient(rgba(15, 90, 80, 0.712), rgb(11, 80, 71));
+}
+
+.grey {
+    background-image: linear-gradient(rgb(37, 45, 48, 0.418), rgb(37, 45, 48));
+    
+}
+
+h1 {
+    font-size: 2.25rem;
```
Comment (Aepera) [inline style.css:36]: We are curious why you choose rem instead of px ?

File: style.css [hunk 11]
```diff
@@ -0,0 +1,302 @@
...
+        justify-content: space-between;
+    }
+
+    .weather img {
+        width: 150px;
+        height: 150px;
+        margin: auto;
+    }
+
+    .temp_city_weather {
```
Comment (Aepera) [inline style.css:168]: Overall, nice and descriptive names for the classes.

**Review by Aepera — APPROVED** [review]
Code review made by Ann-Sofi and Astrid.

**Review by KarolCislowski — APPROVED** [review]
This is a very nice and clean project 🌈 I really like the visual design of it! 🎨 

Great Job!!

**Pull request closed without merging**